### PR TITLE
shrink package size

### PR DESCRIPTION
**Pull request by tlhunter**
### What does this PR do?

- shaves a few kilobytes from the package size

#### Newly Removed Files

```
CONTRIBUTING.md
MIGRATING.md
packages/dd-trace/src/external-logger/test/index.spec.js
scripts/check_licenses.js
scripts/check-proposal-labels.js
scripts/helpers/color.js
scripts/helpers/exec.js
scripts/helpers/title.js
scripts/install_plugin_modules.js
scripts/publish_docs.js
scripts/st.js
```

#### Before

```
=== Tarball Details ===
name:          dd-trace
version:       6.0.0-pre
filename:      dd-trace-6.0.0-pre.tgz
package size:  387.2 kB
unpacked size: 1.7 MB
shasum:        df236cccdb022c4c440f09490efeda8b50e4c83c
integrity:     sha512-+WggnCP1goY4a[...]orLmdoDJXy2lg==
total files:   520
```

#### After

```
=== Tarball Details ===
name:          dd-trace
version:       6.0.0-pre
filename:      dd-trace-6.0.0-pre.tgz
package size:  374.0 kB
unpacked size: 1.7 MB
shasum:        61f7fde6a4c9884487f7d06f9ed16868c2d9640d
integrity:     sha512-6HC00Pgr3xSIb[...]s8wcwF6eWHlBA==
total files:   509                              
```

### Motivation
- smaller is always better

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 !packages/*/src/**/*
 !packages/*/index.js
-!scripts/**/*
+!scripts/preinstall.js
 !vendor/**/*
-!CONTRIBUTING.md
 !LICENSE
 !LICENSE-3rdparty.csv
-!MIGRATING.md
-!README.md
```
Comment (Qard) [inline .npmignore:12]: We need the readme in the package for https://www.npmjs.com/package/dd-trace to display something useful.

**Review by Qard — CHANGES_REQUESTED** [review]

File: .npmignore [hunk 2]
```diff
@@ -1,15 +1,13 @@
 *
+packages/dd-trace/src/external-logger/test/index.spec.js
 !ext/**/*
 !packages/*/lib/**/*
 !packages/*/src/**/*
 !packages/*/index.js
-!scripts/**/*
+!scripts/preinstall.js
```
Comment (rochdev) [inline .npmignore:7]: Should this actually be removed? Preinstall scripts are discouraged and can trip security tools.

File: .npmignore [hunk 3]
```diff
@@ -1,15 +1,13 @@
 *
+packages/dd-trace/src/external-logger/test/index.spec.js
```
[the remark this answers is not in the document]
Comment (tlhunter) [reply .npmignore:2]: Whoops, didn't mean to add that line. The problem was fixed after moving the file.

File: .npmignore (hunk 2 above)
Comment (tlhunter) [reply .npmignore:6]: Possibly... Worth having a follow up discussion.

File: .npmignore (hunk 1 above)
Comment (tlhunter) [reply .npmignore:12]: Good point

**Review by rochdev — APPROVED** [review]

File: .npmignore (hunk 2 above)
Comment (Qard) [reply .npmignore:6]: It seems to just be doing what the `engines` field already does...so not really sure what reason there is to keep it. Was there some issue with the `engines` field being disregarded with some package managers or something?

**Review by Qard — APPROVED** [review]

**Pull request merged**